### PR TITLE
fix(medusa): Draft order inventory management

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Adjust inventory when registering payment of a draft order
- only adjust reservations if a location is registered when cancelling a fulfillment
- disregard previously fulfilled quantities when cancelling an order (since now quantities are not adjusted when cancelling fulfillments)

Fixes CORE-1002

### CODE REVIEW

File: packages/medusa/src/services/order.ts [hunk 1]
```diff
@@ -1146,23 +1146,13 @@ class OrderService extends TransactionBaseService {
...
-            return acc
-          }, acc)
-        },
-        {}
-      )
-
       await Promise.all(
         order.items.map(async (item) => {
           if (item.variant_id) {
             return await inventoryServiceTx.deleteReservationsByLineItem(
```
Comment (srindom) [inline packages/medusa/src/services/order.ts:1152]: question: what does this method do? The name and call signature is a bit confusing - maybe worth renaming it. I.e.: Does it update the reservation with the quantity param - or does it reduce it with the quantity param? 

Doesn't have to be part of this PR necessarily

**Review by srindom — APPROVED** [review]
LGTM!

**Comment by rsurjano** [comment]
issue still persist when creating a Draft order

**Comment by pKorsholm** [comment]
Hi @rsurjano 

We decided not to automatically allocate order lines on draft orders. Instead you can either create a custom subscriber or manually allocate line items after the order has been created. The allocations follow to the order once the draft order is marked as payed.

Is this the issue you are referring to?

**Comment by rsurjano** [comment]
> Hi @rsurjano
> 
> We decided not to automatically allocate order lines on draft orders. Instead you can either create a custom subscriber or manually allocate line items after the order has been created. The allocations follow to the order once the draft order is marked as payed.
> 
> Is this the issue you are referring to?

thank you for feedback, that is right

**Pull request merged**